# Problema 4. Accidentes

El número de accidentes que se producen en una vía de circulación de una ciudad
es, en promedio, de dos a la semana. Determínese:

(a) Probabilidad de que no se registre ningún accidente en una semana determina-
da.

(b) Probabilidad de que en un mes haya más de 5 accidentes.

### 1. Identificamos las variables

Tenemos una variable:

*promedio de accidentes por semana*

### 2. Defina el modelo del sistema y los objetivos de la simulación

Sabemos que existe un promedio de dos accidentes por semana, por lo cual se utilizará la función np.random.poisson para generar números aleatorios con una distribución de Poisson con el parámetro lambda correspondiente a la tasa de accidentes por semana o por mes.
Luego, se cuentan los casos donde no hay accidentes en una semana y donde hay más de 5 accidentes en un mes y se dividen por el número total de simulaciones para obtener las probabilidades estimadas.

Objetivo: Determinar la probabilidad de 0 accidentes en una semana o más de 5 en un mes.

In [46]:
import numpy as np
import math

In [47]:
def poisson_probability(k, lamb):
    """Calcula la probabilidad de obtener k eventos en un intervalo de tiempo con una tasa lambda utilizando la distribución de Poisson."""
    return (np.exp(-lamb) * lamb**k) / np.math.factorial(k)

In [48]:
def probability_no_accidents_week(lamb):
    """Calcula la probabilidad de no haber accidentes en una semana con una tasa lambda."""
    return poisson_probability(0, lamb)

Luego, se define la función **probability_no_accidents_week** para calcular la probabilidad de no haber accidentes en una semana con una tasa *lamb*

In [54]:
def probability_more_than_5_accidents_month(lamb):
    """Calcula la probabilidad de haber más de 5 accidentes en un mes con una tasa lambda.""" 
    prob_more_than_5 = 1 - sum([math.exp(-2)*(2*k)/math.factorial(k) for k in range(6)])
    return prob_more_than_5

Se define la función **probability_more_than_5_accidents_month** para calcular la probabilidad de haber más de 5 accidentes en un mes con una tasa *lamb*

### 3. Diseñe el experimento

In [55]:
lambda_week = 2
lambda_month = lambda_week * 4
prob_zero_accidents = probability_no_accidents_week(lambda_week)
prob_more_than_5_accidents = probability_more_than_5_accidents_month(lambda_month)


print("Probabilidad de no haber accidentes en una semana:", round(prob_zero_accidents*100,2))
print("Probabilidad de haber más de 5 accidentes en un mes:", round(prob_more_than_5_accidents*100,2))

Probabilidad de no haber accidentes en una semana: 13.53
Probabilidad de haber más de 5 accidentes en un mes: 26.69


### 4. Analíticamente

(a) La probabilidad de que no se registre ningún accidente en una semana determinada se puede calcular utilizando la distribución de Poisson con un valor medio de dos accidentes por semana:

P(X=0) = e^(-2) * 2^0 / 0! = e^(-2) ≈ 0.1353

Donde X es el número de accidentes en una semana determinada.

Entonces, la probabilidad de no tener ningún accidente en una semana es del 13.53%.

(b) La probabilidad de que en un mes haya más de 5 accidentes se puede calcular sumando las probabilidades de tener 6, 7, 8, 9, 10 o más accidentes en un mes. Utilizando la distribución de Poisson con un valor medio de 2 accidentes por semana, podemos calcular cada una de estas probabilidades y luego sumarlas:

P(X > 5) = P(X=6) + P(X=7) + P(X=8) + P(X=9) + P(X=10) + ...

P(X=k) = e^(-2) * 2^k / k!

Entonces:

P(X > 5) = 1 - P(X <= 5)
= 1 - [P(X=0) + P(X=1) + P(X=2) + P(X=3) + P(X=4) + P(X=5)]
= 1 - [e^(-2) + 2e^(-2) + 4e^(-2)/2 + 8e^(-2)/6 + 16e^(-2)/24 + 32e^(-2)/120]
≈ 0.2650

Lo que significa que hay una probabilidad del 26.5% de que en un mes haya más de 5 accidentes en esa vía de circulación de la ciudad.